src/components/ui/button/ButtonV2.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { FC, HTMLAttributes, ReactNode } from "react"
+import styled, { css } from "styled-components"
+
+export interface ButtonProps
+  extends HTMLAttributes<HTMLButtonElement | HTMLAnchorElement> {
+  children?: ReactNode
+  size?: "small" | "medium"
+  variant?: "default" | "primary" | "secondary"
+  HTMLElement?: "button" | "a"
```

Comment:
You can omit this and use the [as](https://styled-components.com/docs/api#as-polymorphic-prop) from styled-component instead. In order to make sure to not allow anything else apart the 2 allowed tags you can lock that with typescript types

Then in that case you won't need any React component but only Styled component